pkg/sql/logictest/testdata/logic_test/create_index

```diff
@@ -1,4 +1,4 @@
-# LogicTest: default-configs local-mixed-22.2-23.1
```

Comment:
cc: @fqazi this test used to directly mutates the payload column in `system.jobs`. In a `local-mixed-22.2-23.1` the `job_info` table does not exist until the relevant upgrades have been run. Is it true that under this mixed configuration this test will always run at V22_2 i.e. without running any of the V23_1 upgrades? In that case I can keep both the queries and conditionally run them based on the config the test is being run under.